src/lib/tests/fixture_app.hpp

```diff
@@ -39,7 +39,7 @@ class FixtureApp : public homeobject::HomeObjectApplication {
     }
 
     homeobject::peer_id_t discover_svcid(std::optional< homeobject::peer_id_t > const& p) const override;
-    std::string lookup_peer(homeobject::peer_id_t const&) const override { return "test_fixture.com"; }
+    std::string lookup_peer(homeobject::peer_id_t const&) const override { return "http://127.0.0.1:4000"; }
```

Comment:
there are two reasons:
1 now, we use 1-replica raft group for test cases, so we need to provide ip and port for the raft server to startup and  listen. for solo dev , `look_up` will never be called, we can just ignore it.  but for raft repl dev, `look_up` will be called to get the port to listen to. so here we need to provide a real ip address and port. 127.0.0.1 is used for local ip, 4000 is the default port used in homestore for running the leader raft node in the 3-replica test case, and I just take it here.

2 we use folly::uri to parse the endpoint returned by cm, please take a look at here
https://github.com/eBay/HomeObject/blob/68d9b2d70bf7adc21a2dbda6f8be9a3e385d8fd5/src/lib/homestore_backend/hs_homeobject.cpp#L64

for folly::uri, we can not use simple ip:port(127.0.0.1:4000) as the parameter of constructor. instead, we must provide a uri-like address(http://127.0.0.1:4000), so that the host and port can be extracted correctlly. so, it is the requirement of folly::uri

Reply:
OK, then we have a problem here with `lookup` integration.

CM will response back with what we registered (in `addNode`),like
```
[01/16/24 20:35:21.349611-07:00] [info] [41] [CmServer.cpp:183:register_node] req: svc_info {
  svc_id {
    name: "StorageManager"
  }
  fqdn: "unknown"
  ip_v4: "10.84.0.238"
  ctrl_svc_port: 10000
  data_svc_port: 10001
  location {
    region: "unknown"
    az: "unknown"
    fd: "unknown"
    asset: "unknown"
    tess_cluster_id: "unknown"
  }
  service_version {
    minor_ver: 2
  }
}
```

And SM will return the "ip_v4" field to `HomeObj` as the return of 
`        auto endpoint = _ho_application->lookup_peer(uuid);`

So we missed both `schema` and `port`

I kind like believe the proper fix should be 

1.  SM change the behavior to add `ctrl_svc_port` into the return, i.e changing the https://github.corp.ebay.com/SDS/storage_mgr/blob/main/src/clients/cm_client/cm_client.cpp#L77 to `ip_v4 + ":"  + ctrl_svc_port`

2.  HomeObj add a dummy "http://" in https://github.com/eBay/HomeObject/blob/68d9b2d70bf7adc21a2dbda6f8be9a3e385d8fd5/src/lib/homestore_backend/hs_homeobject.cpp#L64. with only purpose  to allow us to use `folly::Uri`, or maybe we handle the URI decoding by ourself as we only spit host/IP.

Reply:
>So we missed both schema and port
yes, that is exactly what I want to do. i want to see what CM will return today, but ...., CM does not work!

1 I do not think we should use ctrl_svc_port as the return port, i think we should use a seperate port dedicated for nuraft_mesg and add it to sm config file and register it to CM. actually, `look_up` is used as follows:
- find the port for raft server itself to initialize nuraft mesage
- find the raft peer (ip:port) to create raft client and leader will use this client to send message to follower.
 
 i agree to do some change at CM side to return ip:port.

2 yes, i also just want to add  "http://" as the prefix. it`s simple.

Reply:
1.  Yes we should use `hr_svc_port` https://github.corp.ebay.com/SDS/nuobject_proto/blob/main/src/proto/sm_svc.proto#L35

`find the port for raft server itself to initialize nuraft mesage`
It would be better to pass-down from SM->HO->HS as an option but it is fine, as now the path is 
`SM--<registration>---> CM----<lookup> ---> HS`

I will create issue for SM to fix the `registration` and `lookup_peer`


2. Would be better to do that in HO side as it is HO choose to use `folly::Uri` and create the issue.

As a summary,  
Could we do 2) and change this line to `127.0.0.1:4000` ?

Reply:
sure ,look good